src/storage/txn/actions/check_txn_status.rs

```diff
@@ -24,7 +24,13 @@ pub fn check_txn_status_lock_exists(
     caller_start_ts: TimeStamp,
     force_sync_commit: bool,
     resolving_pessimistic_lock: bool,
+    verify_is_primary: bool,
 ) -> Result<(TxnStatus, Option<ReleasedLock>)> {
+    if verify_is_primary && !primary_key.is_encoded_from(&lock.primary) {
+        // Return the current lock info to tell the client what the actual primary is.
+        return Err(ErrorInner::KeyIsLocked(lock.into_lock_info(primary_key.into_raw()?)).into());
```

Comment:
I'm wondering whether the semantics of `KeyIsLocked` is clear and explicit. How about defining a new kind of error?

Reply:
Ok. Please take a look at this https://github.com/pingcap/kvproto/pull/1107